### PR TITLE
move resource options from Config to hostConfig

### PR DESCRIPTION
**Pull request by hqhq**
Cgroup resources are host dependent, they should be in hostConfig.

For backward compatibility, we just copy it to hostConfig, and leave it in
Config for now, so there is no regressions, but the right way to use this
throught json is to put it in HostConfig, like:
  {
      "Hostname": "",
      ...
      "HostConfig": {
          "Memory": 314572800,
      ...
      }
  }

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

### CODE REVIEW

**Comment by hqhq** [comment]
@crosbymichael @unclejack @LK4D4 please take a look at this one, once this is good to go, the movement for other options will on their way soon.

Changes for test cases will in subsequent PR, these changes won't made any test cases fail.

**Comment by LK4D4** [comment]
@hqhq cool, thank you! But I think that this will be in 1.18 api version and we're now in sort of "code freeze"
@crosbymichael What should we do?

**Comment by crosbymichael** [comment]
approach looks good to me.  Maybe we should do all the fields in one PR so we get everything updated in one merge.

**Comment by SvenDowideit** [comment]
and something in the release notes and API recent changes for those that were using the API.

**Comment by hqhq** [comment]
updated, add all resource option changes in this PR.

@SvenDowideit do you mean add something for `docs/sources/release-notes.md`?
Shouldn't this be changed when we really release a version? Or do I need to do something in this PR?

**Comment by SvenDowideit** [comment]
I prefer to keep all related updates in one PR, that way they hit the repo together, and can't be forgotten. but tbh, I'm not sure what the process is.

so this PR is only about moving and renaming, and then #9536 adds better help?

in that case, Docs LGTM @fredlf @jamtur01

**Comment by jamtur01** [comment]
Docs LGTM

**Comment by hqhq** [comment]
ping @crosbymichael @LK4D4

**Comment by unclejack** [comment]
@hqhq You've got to rebase the PR because it can't be merged.

**Comment by hqhq** [comment]
@unclejack @crosbymichael @LK4D4 
Rebased.

File: docs/sources/reference/api/docker_remote_api_v1.17.md [hunk 1]
```diff
@@ -180,7 +180,8 @@ Json Parameters:
 -   **MemorySwap**- Total memory usage (memory + swap); set `-1` to disable swap.
 -   **CpuShares** - An integer value containing the CPU Shares for container
       (ie. the relative weight vs othercontainers).
-    **CpuSet** - String value containg the cgroups Cpuset to use.
+-   **Cpuset** - The same as CpusetCpus, but obsoleted, please don't use.
```
Comment (fredlf) [inline docs/sources/reference/api/docker_remote_api_v1.17.md]: obsoleted --> deprecated

**Comment by fredlf** [comment]
A couple of copy-edits, otherwise docs LGTM.

**Comment by SvenDowideit** [comment]
I presume these API changes need to move to the 1.18 API doc

File: docs/sources/reference/api/docker_remote_api_v1.17.md (hunk 1 above)
Comment (hqhq) [inline docs/sources/reference/api/docker_remote_api_v1.17.md]: Done.

**Comment by cpuguy83** [comment]
Why are we changing the CLI flag?

**Comment by jfrazelle** [comment]
ping @ashahab-altiscale for the lxc template change

**Comment by ashahab-altiscale** [comment]
@jfrazelle Thanks for letting me know. Can you start an lxc build on this?

**Comment by hqhq** [comment]
@cpuguy83 Do you mean we change --cpuset to --cpuset-cpus?
Because it's a better name, cpuset is not only for cpuset.cpus, but also for cpuset.mems, I was about to add support for cpuset.mems, by then, the name --cpuset would not be properly.

**Comment by hqhq** [comment]
Rebased and move the API change to v1.18 API doc.

**Comment by cpuguy83** [comment]
@hqhq ah, ok.  Also didn't notice originally that you did actually deprecate the old syntax properly.
So this LGTM.

Probably want to go ahead and squash commits.

**Comment by vbatts** [comment]
LGTM

**Comment by jfrazelle** [comment]
LGTM just needs commits to be squashed :) and maybe @crosbymichael because he is very good at catching weird things in the configs

**Comment by hqhq** [comment]
Hi, sorry for the late, I'm not sure how these commits can be squashed any more, are you proposing to make all the changes in one commit?
I think it's better to do one thing in one commit, if you think it's better to take the whole movement as one thing, it's ok for me to squash it, just let me know what you think :)

**Comment by hqhq** [comment]
Hi, this PR is depended by a lot of PRs, can we pick this up so we can move on?
@crosbymichael

**Comment by hqhq** [comment]
ping.
@crosbymichael @jfrazelle @LK4D4

**Comment by hqhq** [comment]
@crosbymichael @jfrazelle @LK4D4 @cpuguy83 
Rebased again, I squash all commits into one commit, hope that's what you want.

Please review this, it have been taken too long (more than 4 months since the original PR), a lot of cgroup improvement are depending on this one, hope we can make some progress soon. Thanks.

**Comment by hqhq** [comment]
ping @LK4D4 @crosbymichael @jfrazelle

**Comment by icecrime** [comment]
As far as I can tell, this PR has both code and docs LGTM, and the commits have been squashed as requested. Thanks for you patience @hqhq, I'm merging this hoping that nobody will have any objections.

**Pull request merged**